enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,260 @@
+on control plane machines are [configured](https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-keepalived.yaml)
+with a lower [VRRP](https://en.wikipedia.org/wiki/Virtual_Router_Redundancy_Protocol)
+priority. This ensures that the API on the control plane nodes is fully
+functional before the API VIP moves.
+
+#### API load balancing
+
+Once the API VIP has moved to one of the control plane nodes, traffic sent from
+external clients to this VIP first hits an `haproxy` load balancer running on
+that control plane node.
+These [instances](https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/baremetal/files/baremetal-haproxy.yaml)
+of `haproxy` are [configured](https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/baremetal/files/baremetal-haproxy-haproxy.yaml)
+to load balance the API traffic across all of the control plane nodes.
+
+### Internal DNS
+
+Externally resolvable DNS records are required for:
+
+* `api.$cluster_name.$base-domain` -
+* `*.apps.$cluster_name.$base_domain` -
+
+In addition, internally resolvable DNS records are required for:
+
+* `api-int.$cluster_name.$base-domain` -
+* `etcd-NNN.$cluster_name.$base-domain` -
+* SRV record: `_etcd-server-ssl._tcp.$cluster_name.$base-domain` - Lists the
+  `etcd-NNN` records.
+
+In cluster networking infrastructure, the goal is is to automate as much of the
+DNS requirements internal to the cluster as possible, leaving only a
+small amount of public DNS configuration to be implemented by the user
+before starting the installation process.
+
+In a non-cloud environment, we do not know the IP addresses of all hosts in
+advance.  Those will come from an organization’s DHCP server.  Further, we can
+not rely on being able to program an organization’s DNS infrastructure in all
+cases.  We address these challenges in two ways:
+
+1. Self host some DNS infrastructure to provide DNS resolution for records only
+   needed internal to the cluster.
+2. Make use of mDNS (Multicast DNS) to dynamically discover the addresses of
+   hosts that we must resolve records for.
+
+#### api-int hostname resolution
+
+The CoreDNS server performing our internal DNS resolution includes
+configuration to resolve the `api-int` hostname. `api-int` will be resolved to
+the API VIP.
+
+The [bootstrap](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/coredns.yaml)
+and [control plane](https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-coredns.yaml)
+instances of CoreDNS run as static pod.
+
+#### mdns-publisher
+
+https://github.com/openshift/mdns-publisher
+
+The `mdns-publisher` is the component that runs on each host to make itself
+discoverable by other hosts in the cluster.  Control plane hosts currently
+advertise both `etcd-NN` and `master-NN` names, and the worker nodes advertise
```

Comment:
would like to know how the publisher get the idx of the machine resp to other node in the cluster.